gladys-assistant/docker-compose.yml

```
@@ -0,0 +1,20 @@
+version: "3.7"
+
+services:
+  web:
+    image: gladysassistant/gladys:v4
```

```suggestion
    image: gladysassistant/gladys:v4@sha256:f694d49bf57426cfc37590eb71a19b0e183cd3e8cbae966ef093bb3d199d318b
```

It would be better to add a digest, so that the exact Dockerfile contents are pinned, not just the tag name.

---

Thanks @highghlow 🙏.

@cicoub13 the main reason to do this is that tags can be "overwritten" on docker hub by just pushing another image to the same tag (e.g., v4). If for some reason a newer image that breaks something gets accidentally or maliciously pushed to the same tag, then Docker will download this new image when new users install the app or when existing users restart their app.

Docker gives the highest priority to the `digest`, which locks-in the exact image referenced by the digest. So even if the v4 tag changes at some point, Docker will still download the image associated with the digest we have tested here.

---

If the recommendation is to fix the image with a hash, then do Pull Requests to update the image, I will follow it.

---

Will do. I'll do this one as well to make things easier: https://github.com/getumbrel/umbrel-apps/pull/1044#discussion_r1606339100

---

multi-arch image digest added here: https://github.com/getumbrel/umbrel-apps/pull/1044/commits/c20b58509b008748452d7aca6b30e2047a14bc8b

An easy way to grab the digest when you know your image is multi-architecture is to run 
```
docker pull <image>:<tag>
```

and you will see the multi-arch digest in the output. For example:
```
$ sudo docker pull gladysassistant/gladys:v4
v4: Pulling from gladysassistant/gladys
...
Digest: sha256:f694d49bf57426cfc37590eb71a19b0e183cd3e8cbae966ef093bb3d199d318b
Status: Downloaded newer image for gladysassistant/gladys:v4
docker.io/gladysassistant/gladys:v4
```